plugins/techdocs-react/src/helpers.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { CompoundEntityRef } from '@backstage/catalog-model';
+
+/**
+ * Lower-case entity triplets by default, but allow override.
+ *
+ * @public
+ */
+export function toLowercaseEntityRefMaybe(
+  entityRef: CompoundEntityRef,
+  config: Config,
+) {
+  return config.getOptionalBoolean(
+    'techdocs.legacyUseCaseSensitiveTripletPaths',
```

Comment:
> _what do you think about adding a check to the reader page to see whether this is set, and if it is, fetch the entity from the catalog to check its casing?_

This could work in theory?  We already make a call to the Catalog on the reader, we'd just have to sequence it if the legacy flag was set.  I think the path right now is

- Entity ref read from `useParams()` [in `<TechDocsReaderPage />`](https://github.com/backstage/backstage/blob/ee60ad9ea48150ddf013f12c1afc53e213bd0420/plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.tsx#L86) and then passed down as props to `<TechDocsReaderPageProvider />`
- Ref gets passed down further via context, but we also [read entity metadata here](https://github.com/backstage/backstage/blob/ee60ad9ea48150ddf013f12c1afc53e213bd0420/plugins/techdocs-react/src/context.tsx#L110-L121) and could maybe look to pass down a modified `entityRef` if case sensitivity is needed.